Kernel/Arch/i386/CPU.cpp

```diff
@@ -1390,6 +1530,9 @@ extern "C" void enter_thread_context(Thread* from_thread, Thread* to_thread)
     set_fs(to_tss.fs);
     set_gs(to_tss.gs);
 
+    read_debug_registers_into(from_thread->debug_register_state());
```

Comment:
Should we guard this case so it's only performed if the processes involved are being traced?